src/optimizer/join_order/join_order_optimizer.cpp

```diff
@@ -768,9 +774,19 @@ JoinOrderOptimizer::GenerateJoins(vector<unique_ptr<LogicalOperator>> &extracted
 		result_relation = node->set;
 		result_operator = std::move(extracted_relations[node->set->relations[0]]);
 	}
-	result_operator->estimated_cardinality = node->GetCardinality<idx_t>();
-	result_operator->has_estimated_cardinality = true;
 	result_operator->estimated_props = node->estimated_props->Copy();
+	result_operator->estimated_cardinality = result_operator->estimated_props->GetCardinality<idx_t>();
+	result_operator->has_estimated_cardinality = true;
+	if (result_operator->type == LogicalOperatorType::LOGICAL_FILTER &&
+	    result_operator->children[0]->type == LogicalOperatorType::LOGICAL_GET) {
+		// FILTER on top of GET, add estimated properties to both
+		auto &filter_props = *result_operator->estimated_props;
+		auto &child_operator = *result_operator->children[0];
+		child_operator.estimated_props = make_unique<EstimatedProperties>(
+		    filter_props.GetCardinality<double>() / CardinalityEstimator::DEFAULT_SELECTIVITY, filter_props.GetCost());
```

Comment:
Why do we need estimated properties on both? If I'm not mistaken, the properties of a join order node are just pulled up from all the nodes below it. Those properties are then used by the JOO.

Or is this so prevent confusion when we print join plans?

Reply:
This is purely to improve `EXPLAIN ANALYZE` output.